File: Archive/Stroika_FINAL_for_STERL_1992/Applications/Tiny/Sources/Tiny.r
/* Copyright(c) Sophist Solutions Inc. 1990-1992.  All rights reserved */

/*
 * $Header: /fuji/lewis/RCS/Tiny.r,v 1.2 1992/09/01 18:00:14 sterling Exp $
 *
 * TODO:
 *
 * Changes:
 *	$Log: Tiny.r,v $
 *		Revision 1.2  1992/09/01  18:00:14  sterling
 *		*** empty log message ***
 *
 *		Revision 1.1  1992/06/20  18:40:22  lewis
 *		Initial revision
 *
 *		Revision 1.1  1992/05/01  23:15:01  lewis
 *		Initial revision
 *
 *		Revision 1.5  92/02/18  17:15:14  17:15:14  lewis (Lewis Pringle)
 *		*** empty log message ***
 *		
 *		Revision 1.4  1992/02/17  23:44:15  lewis
 *		Use new version support.
 *
 *
 *
 */


#include	"ApplicationVersion"


resource 'vers' (1) {
	kMajorVersion,
	kMinorVersion,
	kStage,
	kStageVersion,
	verUS,
	kShortVersionString,
	kLongVersionString,
};



resource 'SIZE' (-1) {
	dontSaveScreen,
	acceptSuspendResumeEvents,
	enableOptionSwitch,
	canBackground,
	multiFinderAware,
	backgroundAndForeground,
	dontGetFrontClicks,
	ignoreChildDiedEvents,
	is32BitCompatible,
	notHighLevelEventAware,
	onlyLocalHLEvents,
	notStationeryAware,
	dontUseTextEditServices,
	reserved,
	reserved,
	reserved,
#if		qDebug
	2048	*	1024,						/* preferred mem size	*/
	1024	*	1024						/* minimum mem size		*/
#else	qDebug
	1024	*	1024,						/* preferred mem size	*/
	512		*	1024						/* minimum mem size		*/
#endif	qDebug
};


type qCreator as 'STR ';
resource qCreator (0) {
	"Tiny"
};



resource 'BNDL' (128) {
	qCreator,
	0,
	{
		'ICN#',
		{
			0, 128
		},
		'FREF',
		{
			0, 128
		}
	}
};




resource 'FREF' (128) {
	'APPL',
	0,
	"Application"
};



resource 'ICN#' (128) {
	{
		/* [1] */
		$"0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000"
		$"0000 0000 FE0E 0000 1010 0000 10E0 0C00 1080 1000 1080 20F8 1040 234C 107C 4040"
		$"1020 4040 1020 2040 0820 2040 0820 1040 003E 1040 0003 1040 0000 7040",
		/* [2] */
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
		$"FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF"
	}
};




/* For gnuemacs:
 * Local Variables: ***
 * mode:C ***
 * tab-width:4 ***
 * End: ***
 */

File: Archive/Stroika_FINAL_for_STERL_1992/Applications/Tiny/Sources/Tiny.cc
/* Copyright(c) Sophist Solutions Inc. 1990-1992.  All rights reserved */

/*
 * $Header: /fuji/lewis/RCS/Tiny.cc,v 1.3 1992/09/08 18:05:28 lewis Exp $
 *
 * TODO:
 *
 * Changes:
 *	$Log: Tiny.cc,v $
 *		Revision 1.3  1992/09/08  18:05:28  lewis
 *		Renamed NULL -> Nil.
 *
 *		Revision 1.2  1992/09/01  18:00:14  sterling
 *		*** empty log message ***
 *
 *		Revision 1.11  1992/02/29  02:23:27  lewis
 *		Override app dtor to remove menus from menubar.
 *
 *
 */

#include	"Debug.hh"

#include	"Tiny.hh"



/*
 ********************************************************************************
 ************************* TinyCommandNamesBuilder ******************************
 ********************************************************************************
 */
TinyCommandNamesBuilder::TinyCommandNamesBuilder ()
{
	/*
	 * Add any command name to the command name table (database) that we define for
	 * our application.
	 */

#if		!qMPW_STATIC_ARRAY_WITH_GEN_INIT_CODE_GEN_BUG
	static	const
#endif
	CmdNameTablePair	kCommandNames[] = {
		CmdNameTablePair (CommandHandler::eAboutApplication, "About Tiny"),
	};
	CommandNameTable::Get ().AddPairs (kCommandNames, sizeof (kCommandNames)/sizeof (kCommandNames[0]));
}


/*
 ********************************************************************************
 ********************************** TinyApplication *****************************
 ********************************************************************************
 */
TinyApplication*	TinyApplication::sThis		=	Nil;

TinyApplication::TinyApplication (int argc, const char* argv []):
	Application (argc, argv),
	fMenuBar (),
	fSystemMenu (),
	fFileMenu (),
	fEditMenu ()
#if		qDebug
	,fDebugMenu ()
#endif
{
	Assert (sThis == Nil);
	sThis = this;

#if		qMacToolkit 
	fMenuBar.AddMenu (&fSystemMenu, fSystemMenu.DefaultName ());
#endif
	fMenuBar.AddMenu (&fFileMenu, fFileMenu.DefaultName ());
	fMenuBar.AddMenu (&fEditMenu, fEditMenu.DefaultName ());
#if		qDebug 
	fMenuBar.AddMenu (&fDebugMenu, fDebugMenu.DefaultName ());
#endif

	SetMenuBar (&fMenuBar);
}

TinyApplication::~TinyApplication ()
{
	SetMenuBar (Nil);

	/*
	 * Remove the menus from the menu bar, since the menus are about to be destroyed, and
	 * it is bad to have bogus pointers in our menu list!!!
	 * (An astute student of C++ might note that in the header we declare the menubar before
	 * the Menus themselves, and so menus will be destroyed first. As a practical matter
	 * we could reverse their order, but that would be too subtle. Further, we might note
	 * that in destroying a menubar, it probably does not refer to any of its menus, and
	 * so their being bogus pointers during this process might lead to no ills, it is clearly
	 * inadvisable practice, and this code is meant to illustrate safe practices as much
	 * as it is meant to work).
	 */
#if		qMacToolkit 
	fMenuBar.RemoveMenu (&fSystemMenu);
#endif
	fMenuBar.RemoveMenu (&fFileMenu);
	fMenuBar.RemoveMenu (&fEditMenu);
#if		qDebug 
	fMenuBar.RemoveMenu (&fDebugMenu);
#endif
}

TinyApplication&	TinyApplication::Get ()
{
	EnsureNotNil (sThis);
	return (*sThis);
}


// For gnuemacs:
// Local Variables: ***
// mode:C++ ***
// tab-width:4 ***
// End: ***
